### PR TITLE
[plugin.video.tyt] 3.0.0

### PR DESCRIPTION
**Pull request by lottaboost**
Rewrote plugin to support redesigned website.

### CODE REVIEW

File: plugin.video.tyt/pyjwt/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+language: python
```
Comment (enen92) [inline plugin.video.tyt/pyjwt/.travis.yml]: Same for this one

File: plugin.video.tyt/main.py [hunk 2]
```diff
@@ -19,247 +18,127 @@
...
   dialog.textviewer(label, text)
 
-def jw_website(page):
-  f = urllib2.urlopen(page)
-  return f.read()
-  
-def sendResponse(cookies, pagename): 
-  conn = httplib.HTTPSConnection("tytnetwork.com")
+def sendResponse(cookies, pagename):
+  conn = httplib.HTTPSConnection("tyt.com")
```
Comment (enen92) [inline plugin.video.tyt/main.py]: minor suggestion: usually a good practice to have the server on a variable you can change without messing with the main code

File: plugin.video.tyt/pyjwt/AUTHORS [hunk 3]
```diff
@@ -0,0 +1,29 @@
+PyJWT lead developer
```
Comment (enen92) [inline plugin.video.tyt/pyjwt/AUTHORS]: this would be better as a script.module if you have not introduced any changes to the module

Comment (enen92) [reply plugin.video.tyt/pyjwt/AUTHORS]: Also note the library itself may be licensed under a different license than your plugin

File: plugin.video.tyt/main.py [hunk 4]
```diff
@@ -19,247 +18,127 @@
...
-          page = jw_website("http:" + video_url)
-          video_url = type1.get_jw2(page)
-          play_video(video_url)
+    if params['action'] == 'episodes':
+      list_episodes(sendResponse(cookie, params['show'] + "?page=" + params['page']), params['show'], params['page'])
+    elif params['action'] == 'play':
+      page = sendResponse(get_cookie(), params['video'])
+      hd, sd = scrape.Watch_Episode(page)
+      if hd is not None:
+          print(hd)
```
[the remark this answers is not in the document]
Comment (lottaboost) [reply plugin.video.tyt/main.py]: fixed

File: plugin.video.tyt/pyjwt/.coveragerc [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[run]
```
[the remark this answers is not in the document]
Comment (lottaboost) [reply plugin.video.tyt/pyjwt/.coveragerc]: fixed

File: plugin.video.tyt/pyjwt/.travis.yml (hunk 1 above)
Comment (lottaboost) [reply plugin.video.tyt/pyjwt/.travis.yml]: fixed

File: plugin.video.tyt/main.py (hunk 2 above)
Comment (lottaboost) [reply plugin.video.tyt/main.py]: Left as is.  Unlikely that the website will change.

File: plugin.video.tyt/pyjwt/AUTHORS (hunk 3 above)
Comment (lottaboost) [reply plugin.video.tyt/pyjwt/AUTHORS]: I would like to do this.  Is there somewhere you can point me to that will give an example of how to do this?

Thanks For the help

Comment (enen92) [reply plugin.video.tyt/pyjwt/AUTHORS]: Yes thats it. Then you just import it as a dependency of this addon

Comment (lottaboost) [reply plugin.video.tyt/pyjwt/AUTHORS]: Ok, merged script.module.pyjwt to repo-scripts.  I remove the pyjwt folder from my plugin, and added the script.module.pyjwt import to my addon.xml.  I guess I'll wait until the repo-scripts to propogate so I can test to make sure everything works, then I'll amend my commit with the updated code.

Comment (enen92) [reply plugin.video.tyt/pyjwt/AUTHORS]: @lottaboost did you end up commiting the changes to your branch (this pr)? There's only a single commit from 2 days ago.

Comment (lottaboost) [reply plugin.video.tyt/pyjwt/AUTHORS]: Not yet, going to do now.  Wanted to wait till the library propagated and repo-scripts, that way people don't think the plugin updated to a broken plugin that is missing a dependency.

Comment (lottaboost) [reply plugin.video.tyt/pyjwt/AUTHORS]: I amended the commit.  It is now ready for merge, pending your approval.
Thanks

**Comment by lottaboost** [comment]
I amended the commit. It is now ready for merge, pending your approval.
Thanks

File: plugin.video.tyt/main.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rechi) [reply plugin.video.tyt/main.py]: still is the code

[the remark this answers is not in the document]
Comment (lottaboost) [reply plugin.video.tyt/main.py]: Not sure what happened.. Anyways, it is commented out now, and the commit has been amended

**Comment by Rechi** [comment]
Looks like you removed `fanart.jpg`, `icon.png` and the images in `resources/images` by accident.

File: plugin.video.tyt/main.py [hunk 6]
```diff
@@ -19,247 +18,127 @@
...
-               "Pop Trigger":       {"url":pop_url, "thumb":pop_thumb, "type":"youtube_channel"},
-               "TYT Interviews":    {"url":tytint_url, "thumb":tytint_thumb, "type":"youtube_channel"},
-               "Nerd Alert":        {"url":nerd_url, "thumb":nerd_thumb, "type":"youtube_channel"},
-               "Secular Talk":      {"url":sec_url, "thumb":sec_thumb, "type":"youtube_channel"},
-               "The Point with Ana Kasparian": {"url":point_url, "thumb":point_thumb, "type":"youtube_channel"},
-               "The Majority Report with Sam Seder": {"url":sam_url, "thumb":sam_thumb, "type":"youtube_channel"},
-               "TYT Live Stream":   {"url":tytlive_url, "thumb":tytlive_thumb, "type":"youtube_video"},
-               "ThinkTank":         {"url":thinktank_url, "thumb":thinktank_thumb, "type":"youtube_channel"}}
-menus = {"main":main_cat,"members":members_cat}
+MEDIA_URL = 'special://home/addons/{0}/resources/images/'.format(PLUGIN_ID)
```
Comment (Rechi) [inline plugin.video.tyt/main.py]: I thought resources/images are still used because of this.

**Comment by Rechi** [comment]
Okay, but you have to readd fanart.jpg and icon.png.

File: plugin.video.tyt/main.py (hunk 6 above)
Comment (lottaboost) [reply plugin.video.tyt/main.py]: I didn't realize that line was in there.  I removed it on the commit I did 2 hours ago (After you mentioned the missing directory).

**Comment by lottaboost** [comment]
All the artwork is now downloaded.

**Comment by lottaboost** [comment]
I'm really sorry, is there any way to cancel the merge.  I found something semi-major that I need to fix?  Sorry for the big hassle.

**Pull request merged**